merkletree/merkletreekv_test.go

```diff
@@ -0,0 +1,128 @@
+package merkletree
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/coniks-sys/coniks-go/storage/kv"
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+func TestTreeStore(t *testing.T) {
+	util.WithDB(func(db kv.DB) {
+		key1 := "key1"
+		val1 := []byte("value1")
+		key2 := "key2"
+		val2 := []byte("value2")
+
+		m1, err := NewMerkleTree()
+		if err != nil {
+			t.Fatal(err)
+		}
+		index1 := vrfPrivKey1.Compute([]byte(key1))
+		if err := m1.Set(index1, key1, val1); err != nil {
+			t.Fatal(err)
+		}
+		index2 := vrfPrivKey1.Compute([]byte(key2))
+		if err := m1.Set(index2, key2, val2); err != nil {
+			t.Fatal(err)
+		}
+		m1.recomputeHash()
+
+		wb := db.NewBatch()
+		m1.StoreToKV(1, wb)
+		err = db.Write(wb)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		m2, err := NewMerkleTreeFromKV(db, 1)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if !bytes.Equal(m2.nonce, m1.nonce) ||
+			!bytes.Equal(m2.root.Hash(m1), m1.hash) {
+			t.Fatal("Bad tree construction")
+		}
+
+		ap := m2.Get(index1)
+		if ap.Leaf.Value() == nil {
```

Comment:
Is this the only condition that indicates that ``key1`` cannot be found in the tree? I'm wondering if we should also be checking the leaf node type and the leaf's index here.